### DISCUSSION TITLE
Generic way to use Tailwind (Tailwind CLI vs. PostCSS integration)

### DISCUSSION THREAD

**Discussion by bennettdams** [tailwindlabs/tailwindcss#4975] [Ideas]
I'm using Tailwind in Next.js, Create React App and Vite projects. 
I use Tailwind for 2 years now, but I never took the time to learn PostCSS. I think this is true for a lot of Tailwind users.
Therefore I still struggle with different configurations for different tools with different versions.
Integrating Tailwind with such frameworks is still a hassle - e.g. CRA still doesn't support PostCSS 8, so JIT mode can't be used (at least not without sketchy additional configuration).
JIT mode with Next.js v11 just works without configuring anything external - but sometimes it does not? I think Webpack 5 should be blamed for that, but... I don't know? There are so many changes in build tools/configuration that's it's annoying to keep up with.

My point is: Isn't there a **"generic"** way to use Tailwind (with purge & JIT)?

When I thought about that, I remembered using the Tailwind CLI once. Looking [at the documentation](https://tailwindcss.com/docs/installation#using-tailwind-cli) now, I found this beautiful line:

> If you’d like to compile your CSS with Tailwind without integrating it directly into any sort of build tooling, you can use the Tailwind CLI tool to generate your CSS without configuring PostCSS or even installing Tailwind as a dependency if you don’t want to.

Can someone, briefly, explain why the Tailwind CLI is not the preferred way to set up Tailwind in our projects? 
What are the downsides with using Tailwind CLI instead of integrating the Tailwind configuration with PostCSS in your build tool?

From reading the docs, the only thing that came to my mind is that you probably can't start the CLI and your dev server (like Next.js' `npm run dev`) at the same time/with one command. So for Tailwind to react to changes with JIT, you probably have to start a second process that runs the Tailwind CLI. But I'm totally down to open another tab in my terminal, if I therefore never have to look at a PostCSS integration, broken build tool or additional config ever again.

**Comment by hacknug** [comment]
> Can someone, briefly, explain why the Tailwind CLI is not the prefered way to set up Tailwind in our projects?

I think this is because in most enviroments (maybe not yours, definitely not mine a few years ago) you already have all those tools you mentioned integrated at the framework level (postcss, webpack…).

Nothing wrong with the CLI imho if you want to keep things simple 👍
